CMakeLists.txt

```diff
@@ -243,31 +265,41 @@ endif()
 option(USE_OPTIMIZED_BLAS "Whether or not to use an optimized BLAS library instead of included netlib BLAS" OFF)
 
 # Check the usage of the user provided BLAS libraries
-if(BLAS_LIBRARIES)
-  include(CheckFortranFunctionExists)
-  set(CMAKE_REQUIRED_LIBRARIES ${BLAS_LIBRARIES})
-  CHECK_FORTRAN_FUNCTION_EXISTS("dgemm" BLAS_FOUND)
-  unset(CMAKE_REQUIRED_LIBRARIES)
-  if(BLAS_FOUND)
-    message(STATUS "--> BLAS supplied by user is WORKING, will use ${BLAS_LIBRARIES}.")
+if(USE_OPTIMIZED_BLAS)
+  if(BLAS_LIBRARIES)
```

Comment:
In general: I think this if shouldn't exists. If the user supplied values trust him/her. It is not the responsible of the build description to check valid inputs.

Reply:
I'm following you on that … I just extended existing logic, not changing everything. My dream is to specify `BLAS_LIBS` and have them just used or alternatively the name of a pkg-config package for the same purpose. Actually, I have CMake FindBLAS modifications pending to be able to override the automagic there with a provided pkg-config package name.

Reply:
> My dream is to specify BLAS_LIBS and have them just used or alternatively the name of a pkg-config package for the same purpose

You can always provide a custom `FindBLAS.cmake`  via `CMAKE_MODULE_PATH` where you can do this (if you adhere to variables defined by the inbuild CMake module ). Build descriptions should stay as clean as possible from library logic. This also means using targets instead of library names: e.g. linking against `BLAS::BLAS` (probably not possible due to cmake restrictions? lapack could provide its own module for it. )

Reply:
Well, yes … or patch the upstream one in the build. My goal is to have such stuff upstreamed instead of hacking around it locally. Actually, my whole point is to get rid of the automagic with hardcoded lists in FindBLAS.cmake altogether. If I have to extend it to achieve that, be it that way. It irks me to have a build-system-specific dependency mechanism that duplicates the efforts of pkg-config.

Reply:
> automagic with hardcoded lists in FindBLAS.cmake

I hope your hardcoded list takes care of custom suffixes and prefixes ;)